mobile/library/swift/cxx/EnvoyHTTPFilterFactory+Cxx.swift

```diff
@@ -0,0 +1,273 @@
+      }
+
+      return envoy_filter_trailers_status(
+        status: envoy_filter_trailers_status_t((result[0] as? NSNumber)?.intValue ?? 0),
+        trailers: toEnvoyHeaders(result[1] as? [String: [String]]),
+        pending_headers: pending_headers,
+        pending_data: pending_data
+      )
+    }
+    filter.pointee.set_request_callbacks = { callbacks, context in
+      let filter = Unmanaged<EnvoyHTTPFilter>.fromOpaque(context!).takeUnretainedValue()
+      guard let setRequestFilterCallbacks = filter.setRequestFilterCallbacks else {
+        return
+      }
+
+      let requestFilterCallbacks = EnvoyHTTPFilterCallbacksImpl(callbacks: callbacks)
+      setRequestFilterCallbacks(requestFilterCallbacks)
+    }
+    filter.pointee.set_response_callbacks = { callbacks, context in
+      let filter = Unmanaged<EnvoyHTTPFilter>.fromOpaque(context!).takeUnretainedValue()
+      guard let setResponseFilterCallbacks = filter.setResponseFilterCallbacks else {
+        return
+      }
+
+      let responseFilterCallbacks = EnvoyHTTPFilterCallbacksImpl(callbacks: callbacks)
+      setResponseFilterCallbacks(responseFilterCallbacks)
+    }
+
+    filter.pointee.on_resume_request =
+    { headers, data, trailers, end_stream, stream_intel, context in
+      let filter = Unmanaged<EnvoyHTTPFilter>.fromOpaque(context!).takeUnretainedValue()
+      guard let onResumeRequest = filter.onResumeRequest else {
+        return envoy_filter_resume_status(
+          status: kEnvoyFilterHeadersStatusContinue,
+          pending_headers: headers,
+          pending_data: data,
+          pending_trailers: trailers
+        )
+      }
+      let pendingHeaders = (headers?.pointee).flatMap(toSwiftHeaders)
+      let pendingData = (data?.pointee).flatMap(Data.init)
+      let pendingTrailers = (trailers?.pointee).flatMap(toSwiftHeaders)
+      let result = onResumeRequest(pendingHeaders, pendingData, pendingTrailers, end_stream,
+                                   stream_intel)
+      return envoy_filter_resume_status(
+        status: envoy_filter_resume_status_t((result[0] as? NSNumber)?.intValue ?? 0),
+        pending_headers: toEnvoyHeadersPtr(result[1] as? [String: [String]]),
+        pending_data: toEnvoyDataPtr(result[2] as? Data),
+        pending_trailers: toEnvoyHeadersPtr(result[3] as? [String: [String]])
+      )
+    }
+
+    filter.pointee.on_resume_response =
+    { headers, data, trailers, end_stream, stream_intel, context in
+      let filter = Unmanaged<EnvoyHTTPFilter>.fromOpaque(context!).takeUnretainedValue()
+      guard let onResumeResponse = filter.onResumeResponse else {
+        return envoy_filter_resume_status(
+          status: kEnvoyFilterHeadersStatusContinue,
+          pending_headers: headers,
+          pending_data: data,
+          pending_trailers: trailers
+        )
+      }
+      let pendingHeaders = (headers?.pointee).flatMap(toSwiftHeaders)
+      let pendingData = (data?.pointee).flatMap(Data.init)
```

Comment:
obj-c implementation uses `release_envoy_data` which frees memory allocated by data. We are leaking this memory in the new implementation I think.